core/arch/arm/include/kernel/delay.h

```diff
@@ -44,11 +44,13 @@ static inline uint64_t arm_cnt_us2cnt(uint32_t us)
 
 static inline uint64_t timeout_init_us(uint32_t us)
 {
+	isb();
 	return read_cntpct() + arm_cnt_us2cnt(us);
```

Comment:
Why not use `barrier_read_cntpct()` here too?

Reply:
Sorry, missed that.